src/Code.elm

```diff
@@ -0,0 +1,391 @@
+      -- let|handle|in|where|match|with|cases|->|if|then|else|and|or
+    | ControlKeyword
+      -- forall|->
+    | TypeOperator
+    | BindingEquals
+    | TypeAscriptionColon
+      -- type|ability
+    | DataTypeKeyword
+    | DataTypeParams
+    | Unit
+      -- unique
+    | DataTypeModifier
+      -- `use Foo bar` is keyword, prefix, suffix
+    | UseKeyword
+    | UsePrefix
+    | UseSuffix
+    | HashQualifier HashQualified
+      -- ! '
+    | DelayForceChar
+      -- ? , ` [ ] @ |
+      -- Currently not all commas in the pretty-print output are marked up as DelimiterChar - we miss
+      -- out characters emitted by Pretty.hs helpers like Pretty.commas.
+    | DelimiterChar
+    | Parenthesis
+    | LinkKeyword -- `typeLink` and `termLink`
+      -- [: :] @[]
+    | DocDelimiter
+      -- the 'include' in @[include], etc
+    | DocKeyword
+
+
+
+-- VIEW
+
+
+syntaxTypeToClassName : SyntaxType -> String
+syntaxTypeToClassName sType =
+    case sType of
+        NumericLiteral ->
+            "numberic-literal"
+
+        TextLiteral ->
+            "text-literal"
+
+        BytesLiteral ->
+            "bytes-literal"
+
+        CharLiteral ->
+            "char-literal"
+
+        BooleanLiteral ->
+            "boolean-literal"
+
+        Blank ->
+            "blank"
+
+        Var ->
+            "var"
+
+        Reference _ ->
+            "reference"
+
+        Referent _ ->
+            "referent"
+
+        Op seqOp ->
+            case seqOp of
+                Cons ->
+                    "op cons"
+
+                Snoc ->
+                    "op snoc"
+
+                Concat ->
+                    "op concat"
+
+        Constructor ->
+            "constructor"
+
+        Request ->
+            "request"
+
+        AbilityBraces ->
+            "ability-braces"
+
+        ControlKeyword ->
+            "control-keyword"
+
+        TypeOperator ->
+            "type-operator"
+
+        BindingEquals ->
+            "binding-equals"
+
+        TypeAscriptionColon ->
+            "type-ascription-colon"
+
+        DataTypeKeyword ->
+            "data-type-keyword"
+
+        DataTypeParams ->
+            "data-type-params"
+
+        Unit ->
+            "unit"
+
+        DataTypeModifier ->
+            "data-type-modifier"
+
+        UseKeyword ->
+            "use-keyword"
+
+        UsePrefix ->
+            "use-prefix"
+
+        UseSuffix ->
+            "use-suffix"
+
+        HashQualifier _ ->
+            "hash-qualifier"
+
+        DelayForceChar ->
+            "delay-force-char"
+
+        DelimiterChar ->
+            "delimeter-char"
+
+        Parenthesis ->
+            "parenthesis"
+
+        LinkKeyword ->
+            "link-keyword"
+
+        DocDelimiter ->
+            "doc-delimeter"
+
+        DocKeyword ->
+            "doc-keyword"
+
+
+viewSegment : SyntaxSegment -> Html msg
+viewSegment (SyntaxSegment sType sText) =
+    span [ class (syntaxTypeToClassName sType) ] [ text sText ]
```

Comment:
Linking is not yet supported. Supporting it might mean tying the syntax text and type closer together instead of being 2 separate fields on `SyntaxSegment` - Perhaps `SyntaxSegment` has all the types as variants and they all have a syntax text field 🤔